spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -1399,7 +1399,7 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       CometConf.COMET_EXEC_ENABLED.key -> "true",
       CometConf.COMET_SHUFFLE_ENFORCE_MODE_ENABLED.key -> "true",
       CometConf.COMET_EXEC_ALL_OPERATOR_ENABLED.key -> "true",
-      "spark.sql.extendedExplainProvider" -> "org.apache.comet.ExtendedExplainInfo") {
```

Comment:
So even we don't config it, the test still can pass?

Reply:
Currently it passes because, before Spark 4.0, `supportsExtendedExplainInfo` in `CometTestBase.scala` is always `false`.

Reply:
I meant previously the config name is wrong, so the config doesn't apply at all. So no matter with or without this config, this test passes?

Reply:
Correct, it passes with or without this config for Spark 3.x because the test is disabled by `supportsExtendedExplainInfo` that is `false`.

For Spark 4.0 or any Spark fork that implements the extended info, `supportsExtendedExplainInfo` returns true. Then it fails with the previous wrong config name.

Reply:
Hmm, but we don't run tests against Spark 4.0 currently, right? It sounds a little strange to have a test that can pass no matter what.

Reply:
Maybe we should move this test to Spark 4.0 particular test folder later.

Reply:
There is another part in the same test that is valid for all Spark versions.
Perhaps, we can consider separating them out in the future cc @parthchandra 
Thanks for reviewing

Reply:
Will do that @kazuyukitanimura